### DISCUSSION TITLE
Installation error

### DISCUSSION THREAD

**Discussion by chiararik** [davidfrantz/force#258] [Q&A] [answered]
Hello!

I am trying to install locally FORCE, but at the final step I get this error despite all the dependencies, including landsatlinks, have been previously installed:

```
chiara@DESKTOP-KHATE8I:~/src/force$ sudo make install
Makefile:64: *** "No landsatlinks in PATH, install dependencies!".  Stop.
```

Any suggestion about how to fix this issue?

Thanks!

Chiara

**Comment by ernstste** [comment]
Hi Chiara,

FORCE requires several dependencies to be installed, see here: https://force-eo.readthedocs.io/en/latest/setup/depend.html
The specific dependency you are missing here can be installed with this command: `pip install numpy gsutil git+https://github.com/ernstste/landsatlinks.git`

We strongly reccomend using the pre-built docker images for FORCE. That way you don't have to worry about any dependencies and can be sure you have a properly set up environment.

Stefan

**Reply by chiararik** [reply]
Dear Stefan, 

yes, that dependency was installed properly, but the issue still persist preventing the FORCE installation:

`pip install numpy gsutil git+https://github.com/ernstste/landsatlinks.git
Defaulting to user installation because normal site-packages is not writeable
Collecting git+https://github.com/ernstste/landsatlinks.git
  Cloning https://github.com/ernstste/landsatlinks.git to /tmp/pip-req-build-up1q4z3o
  Running command git clone --filter=blob:none --quiet https://github.com/ernstste/landsatlinks.git /tmp/pip-req-build-up1q4z3o
  Resolved https://github.com/ernstste/landsatlinks.git to commit 635423aa4d4db84faa3f805e7d1ff1759035ec26
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in /usr/lib/python3/dist-packages (1.21.5)
Requirement already satisfied: gsutil in ./.local/lib/python3.10/site-packages (5.17)
Requirement already satisfied: gcs-oauth2-boto-plugin>=3.0 in ./.local/lib/python3.10/site-packages (from gsutil) (3.0)
Requirement already satisfied: crcmod>=1.7 in ./.local/lib/python3.10/site-packages (from gsutil) (1.7)
Requirement already satisfied: google-reauth>=0.1.0 in ./.local/lib/python3.10/site-packages (from gsutil) (0.1.1)
Requirement already satisfied: google-apitools>=0.5.32 in ./.local/lib/python3.10/site-packages (from gsutil) (0.5.32)
Requirement already satisfied: argcomplete>=1.9.4 in ./.local/lib/python3.10/site-packages (from gsutil) (2.0.0)
Requirement already satisfied: pyOpenSSL>=0.13 in ./.local/lib/python3.10/site-packages (from gsutil) (22.1.0)
Requirement already satisfied: six>=1.12.0 in /usr/lib/python3/dist-packages (from gsutil) (1.16.0)
Requirement already satisfied: monotonic>=1.4 in ./.local/lib/python3.10/site-packages (from gsutil) (1.6)
Requirement already satisfied: fasteners>=0.14.1 in ./.local/lib/python3.10/site-packages (from gsutil) (0.18)
Requirement already satisfied: google-auth[aiohttp]>=2.5.0 in ./.local/lib/python3.10/site-packages (from gsutil) (2.15.0)
Requirement already satisfied: httplib2==0.20.4 in ./.local/lib/python3.10/site-packages (from gsutil) (0.20.4)
Requirement already satisfied: retry-decorator>=1.0.0 in ./.local/lib/python3.10/site-packages (from gsutil) (1.1.1)
Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /usr/lib/python3/dist-packages (from httplib2==0.20.4->gsutil) (2.4.7)
Requirement already satisfied: gdal in /usr/lib/python3/dist-packages (from landsatlinks==1.0.0) (3.4.1)
Requirement already satisfied: requests in ./.local/lib/python3.10/site-packages (from landsatlinks==1.0.0) (2.28.1)
Requirement already satisfied: tqdm in ./.local/lib/python3.10/site-packages (from landsatlinks==1.0.0) (4.64.1)
Requirement already satisfied: oauth2client>=2.2.0 in ./.local/lib/python3.10/site-packages (from gcs-oauth2-boto-plugin>=3.0->gsutil) (4.1.3)
Requirement already satisfied: rsa==4.7.2 in ./.local/lib/python3.10/site-packages (from gcs-oauth2-boto-plugin>=3.0->gsutil) (4.7.2)
Requirement already satisfied: boto>=2.29.1 in ./.local/lib/python3.10/site-packages (from gcs-oauth2-boto-plugin>=3.0->gsutil) (2.49.0)
Requirement already satisfied: pyasn1>=0.1.3 in ./.local/lib/python3.10/site-packages (from rsa==4.7.2->gcs-oauth2-boto-plugin>=3.0->gsutil) (0.4.8)
Requirement already satisfied: pyasn1-modules>=0.2.1 in ./.local/lib/python3.10/site-packages (from google-auth[aiohttp]>=2.5.0->gsutil) (0.2.8)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in ./.local/lib/python3.10/site-packages (from google-auth[aiohttp]>=2.5.0->gsutil) (5.2.0)
Requirement already satisfied: aiohttp<4.0.0dev,>=3.6.2 in ./.local/lib/python3.10/site-packages (from google-auth[aiohttp]>=2.5.0->gsutil) (3.8.3)
Requirement already satisfied: pyu2f in ./.local/lib/python3.10/site-packages (from google-reauth>=0.1.0->gsutil) (0.1.5)
Requirement already satisfied: cryptography<39,>=38.0.0 in ./.local/lib/python3.10/site-packages (from pyOpenSSL>=0.13->gsutil) (38.0.4)
Requirement already satisfied: charset-normalizer<3,>=2 in ./.local/lib/python3.10/site-packages (from requests->landsatlinks==1.0.0) (2.1.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./.local/lib/python3.10/site-packages (from requests->landsatlinks==1.0.0) (1.26.13)
Requirement already satisfied: idna<4,>=2.5 in ./.local/lib/python3.10/site-packages (from requests->landsatlinks==1.0.0) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in ./.local/lib/python3.10/site-packages (from requests->landsatlinks==1.0.0) (2022.12.7)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.local/lib/python3.10/site-packages (from aiohttp<4.0.0dev,>=3.6.2->google-auth[aiohttp]>=2.5.0->gsutil) (6.0.3)
Requirement already satisfied: yarl<2.0,>=1.0 in ./.local/lib/python3.10/site-packages (from aiohttp<4.0.0dev,>=3.6.2->google-auth[aiohttp]>=2.5.0->gsutil) (1.8.2)
Requirement already satisfied: frozenlist>=1.1.1 in ./.local/lib/python3.10/site-packages (from aiohttp<4.0.0dev,>=3.6.2->google-auth[aiohttp]>=2.5.0->gsutil) (1.3.3)
Requirement already satisfied: aiosignal>=1.1.2 in ./.local/lib/python3.10/site-packages (from aiohttp<4.0.0dev,>=3.6.2->google-auth[aiohttp]>=2.5.0->gsutil) (1.3.1)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in ./.local/lib/python3.10/site-packages (from aiohttp<4.0.0dev,>=3.6.2->google-auth[aiohttp]>=2.5.0->gsutil) (4.0.2)
Requirement already satisfied: attrs>=17.3.0 in ./.local/lib/python3.10/site-packages (from aiohttp<4.0.0dev,>=3.6.2->google-auth[aiohttp]>=2.5.0->gsutil) (22.1.0)
Requirement already satisfied: cffi>=1.12 in ./.local/lib/python3.10/site-packages (from cryptography<39,>=38.0.0->pyOpenSSL>=0.13->gsutil) (1.15.1)
Requirement already satisfied: pycparser in ./.local/lib/python3.10/site-packages (from cffi>=1.12->cryptography<39,>=38.0.0->pyOpenSSL>=0.13->gsutil) (2.21)`

Ok, I will use the docker image, thanks

**Comment by davidfrantz** [comment, marked as answer]
pip might have installed landsatlinks to a location that is not in the PATH variable.

If this is the case, you need to add that path to $PATH.

**Comment by chiararik** [comment]
I am not very used to bash scripting, I have tried this, but unsuccesfully

`chiara@DESKTOP-KHATE8I:~$ PATH=~./.local/lib/python3.10/site-packages
chiara@DESKTOP-KHATE8I:~/src$ cd force
chiara@DESKTOP-KHATE8I:~/src/force$ make -j
Command 'make' is available in the following places
 * /bin/make
 * /usr/bin/make
The command could not be located because '/usr/bin:/bin' is not included in the PATH environment variable.
make: command not found
chiara@DESKTOP-KHATE8I:~/src/force$ PATH=~/usr/bin:/bin
chiara@DESKTOP-KHATE8I:~/src/force$ make -j
Makefile:64: *** "No landsatlinks in PATH, install dependencies!".  Stop.
chiara@DESKTOP-KHATE8I:~/src/force$`

**Reply by davidfrantz** [reply]
hmm, in this case, I would very much recommend you to use FORCE through Docker.

But for future reference, please see https://linuxize.com/post/how-to-add-directory-to-path-in-linux/

Best,
David

**Reply by gdsea01** [reply]
Hi,
Matt here. 
I had the same issue described by Chiararik .Then, I tried to install FORCE through Docker (several times) and it returned this: 

'Got permission denied while trying to connect to the Docker daemon socket at: unix://var/run/docker.sock: Post "http://%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=davidfrantz%2Fforce&tag=3.6.5": dial unix /var/run/docker.sock: connect: permission denied'

Any suggestion ?

**Reply by ernstste** [reply]
Hi Matt,

this looks like the user account you are using is missing the appropriate permissions.
Make sure the user is added to the docker group. If that is the case already, try logging out and back in again or reboot the machine.
